sdk/identity/identity/src/credentials/environmentCredential.ts

```diff
@@ -34,6 +51,13 @@ export class EnvironmentCredential implements TokenCredential {
    * @param options Options for configuring the client which makes the authentication request.
    */
   constructor(options?: TokenCredentialOptions) {
+    // Keep track of any missing environment variables for error details
+    AllSupportedEnvironmentVariables.forEach((v) => {
```

Comment:
I'm being picky, but what about simplifying this a little?

```ts
this._environmentVarsMissing = AllSupportedEnvironmentVariables.filter(v => {
   return process.env[v] === undefined;
});
```

Reply:
Ahhh, I was using a map/filter approach before but the TypeScript compiler was confused about the output.  This is better!